docs/proposals/file-sync-location.md

```diff
@@ -0,0 +1,26 @@
+
+# File Sync Location for Devfile Components in Odo
+
+## Background 
+
+Currently, odo uses the following logic to determine where to sync source code to a devfile component:
+
+1) If the `sourceMapping` field is set, odo will mount the source volume under to the specified folder and sync the source code there
+
+2) If `sourceMapping` is unset, odo will do one of two things:
+   a) If a single project is defined in the devfile, odo will sync to `/projects/<projectName>`
+   b) If no projects are defined, or there are multiple projects defined, odo will sync to `/projects`
```

Comment:
This problem cloud be solved with "local" components (https://github.com/devfile/kubernetes-api/issues/80) If devfile have support for local component, we could require from devfile to always have at least one project

Reply:
Good to know about.

Reply:
I was wrong :-/ Local component is something else than I thought :-/